enhancements/installer/kubernete-slb-owned-api-loadbalancers.md

```diff
@@ -0,0 +1,508 @@
+
+The `bootstrap` machine hosts the bootstrap-control-plane, and the installer has pre-seeded the DNS entries with reserved ips for loadbalancers to 
+be created by the cluster. The kubelet on `master-{0,1,2}` use the DNS to resolve the endpoints to the `bootstrap` machine, 
+and communicate with the k8s API.
+
+An entity generates Kubernetes service type loadbalancer objects using the static ips created by the installer and pushes them to the cluster using 
+the bootstrap control plane. The bootstrap-control-plane 's kube-controller-manager creates the LBs for the services.
+
+**Stage 2:** Bootstrapping complete
+
+PUBLIC ZONE:api=static-public-lb-ip
+PRIVATE ZONE:api=bootstrap-private-ip,static-private-lb-ip;api-int=bootstrap-private-ip,static-private-lb-ip
+
+public-lb:master-0(green);master-1(green);master-2(green)
+private-lb:master-0(green);master-1(green);master-2(green)
+
+Bootstrapping of the k8s API is complete when the necessary pods are running, and the service type load balancers are running. At this point 
+the `bootstrap-control-plane` shuts itself down. The clients can now reach the k8s API using the loadbalancer ips, the clients trying to reach API 
+using `bootstrap` host will fail and round-robin to the loadbalancer ips.
+
+**Stage 3:** Bootstrap host removed
+
+PUBLIC ZONE:api=static-public-lb-ip
+PRIVATE ZONE:api=static-private-lb-ip;api-int=static-private-lb-ip
+
+public-lb:master-0(green);master-1(green);master-2(green)
+private-lb:master-0(green);master-1(green);master-2(green)
+
+The installer terminates the `bootstrap` machine and also removes the `bootstrap` machine's ip addresses from the DNS. And only the `master-{0,1,2}` machines
+remain with only loadbalancer ips in the DNS.
+
+### User Stories
+
+#### Readiness logic maintained by apiserver team
+
+Today the load balancers are created by the installer and therefore the health-checks endpoints and levels are controlled by the installer, 
+but with SLB owned API loadbalancers the apiserver team can defined the readiness on the pods themselves and the k8s-networking-provider will 
+bring pods in/out of backend.
+
+#### Changing the publising strategy of API as day-2
+
+Since the API loadbalancers are now Service objects, a user can easily remove the public facing object to remove the Internet facing access of the API,
+or add Internet facing access for internal clusters.
+
+#### API endpoints tracking
+
+The ability of the kubelet to communicate with k8s API is very important for self hosted control-plane. Currently the kubelet uses the load balancer to 
+communicate with the API, which has seen various bugs in the past causing the kubelet to loose connection to API.
+
+To improve the resilency of the connection, we can program alternate endpoints for the API like localhost, individual pod/node ip etc, allowing the kubelets 
+to use multiple paths to reach the API.
```

Comment:
We stopped doing this for kcm+ks. It turned out to be even less reliable because localhost has no guaranteed life-cylce (graceful termination) as long as you don't track `/readyz`. So these services are never warned of termination of the API.